src/config/jobconfig.ts

```diff
@@ -19,172 +19,236 @@ import { ApiProperty } from "@nestjs/swagger";
 import { JobClass } from "../jobs/schemas/job.schema";
 import { CreateJobDto } from "../jobs/dto/create-job.dto";
 import { UpdateJobStatusDto } from "../jobs/dto/update-jobstatus.dto";
-
+import { oneOrMore } from "../common/utils";
 
 /**
  * Encapsulates all responses to a particular job type (eg "archive")
  */
 export class JobConfig {
-    jobType: string;
-
-    create: JobCreateAction[];
-    // read: JobReadAction[];
-    update: JobUpdateAction[];
-
-    constructor(jobType: string, create: JobCreateAction[]=[], read=[], update: JobUpdateAction[]=[]) {
-        this.jobType = jobType;
-        this.create = create;
-        // this.read = read;
-        this.update = update;
-    }
-
-    /**
-     * Parse job configuration json by dispatching to currently registered JobActions
-     * @param data JSON
-     * @returns
-     */
-    static parse(data: Record<string, any>) {
-        const type = data["jobType"];
-        const create = "create" in data ? oneOrMore(data["create"]).map((json) => parseCreateAction(json["action"])) : [];
-        const read = undefined; //"read" in data ? oneOrMore(data["read"]).map((json) => parseReadAction(json["action"])) : [];
-        const update = "update" in data ? oneOrMore(data["update"]).map((json) => parseUpdateAction(json["action"])) : [];
-        return new JobConfig(type, create, read, update);
-    }
+  jobType: string;
+
+  create: JobOperation<CreateJobDto>;
+  // read: JobReadAction[];
+  update: JobOperation<UpdateJobStatusDto>;
+
+  constructor(
+    jobType: string,
+    create: JobOperation<CreateJobDto>,
+    read = undefined,
+    update: JobOperation<UpdateJobStatusDto>,
+  ) {
+    this.jobType = jobType;
+    this.create = create;
+    // this.read = read;
+    this.update = update;
+  }
+
+  /**
+   * Parse job configuration json by dispatching to currently registered JobActions
+   * @param data JSON
+   * @returns
+   */
+  static parse(data: Record<string, any>) {
+    const type = data["jobType"];
+    const create = JobOperation.parse<CreateJobDto>(
```

Comment:
Is this a good spelling for this concept? We were struggling to find a good term for jobType ('archive'), operator ('update') and actionType ('email').

Reply:
How about this:
- Job type: archive
- Job phase: create, update, post
- Action type: email, log or else
A job phase can have one or more actions.